web/reader.js

```diff
@@ -16,8 +16,101 @@ var PDFNavigator = cozy.Control.Navigator.extend({
   EOT: true
 });
 
+var PDFSearch = cozy.Control.Search.extend({
+
+  onAdd: function(reader) {
+    var self = this;
+    console.log("AHOY AHOY", this);
+    this._reader.on('ready', function() {
+      // updatefindcontrolstate -- emites state=FOUND before all the results are found
+      this._reader.PDFViewerApplication.eventBus.on('updatefindmatchescount', this._waitForResults.bind(this));
```

Comment:
Search in PDFViewerApplication is asynchronous --- because of the way the UI is designed in the default viewer, results can be gathered lazily. Trying to use the events to wait until all the matches have been located.